atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -111,6 +116,10 @@
 
     protected final TransactionKnowledgeComponents knowledge;
 
+    // unsure about if using `Transaction` as key is janky - an alternative is adding a UUID (or an ID counter) to
+    // transactions we are tracking (OpenTransactionImpl for now) and using that
+    protected final ConcurrentMap<Transaction, Long> transactionClockStartTs = new ConcurrentHashMap<>();
```

Comment:
```suggestion
    protected final Map<Transaction, Stopwatch> transactionClock = new ConcurrentHashMap<>();
```
Transaction as key is fine in this case as we really do want to key on the reference